internal/pkg/agent/cmd/run.go

```diff
@@ -557,3 +559,39 @@ func setupMetrics(
 func isProcessStatsEnabled(cfg *monitoringCfg.MonitoringConfig) bool {
 	return cfg != nil && cfg.HTTP.Enabled
 }
+
+// handleUpgrade checks if agent is being run as part of an
+// ongoing upgrade operation, i.e. being re-exec'd and performs
+// any upgrade-specific work, if needed.
+func handleUpgrade() error {
+	upgradeMarker, err := upgrade.LoadMarker()
+	if err != nil {
+		return fmt.Errorf("unable to load upgrade marker to check if Agent is being upgraded: %w", err)
+	}
+
+	if upgradeMarker == nil {
+		// We're not being upgraded. Nothing more to do.
+		return nil
+	}
+
+	// In v8.8.0, we introduced a new installation marker file to indicate that
+	// an Agent was running as installed. When an installed Agent that's older
+	// than v8.8.0 is upgraded, this installation marker file is not present.
+	// So, in such cases, we need to create it manually post-upgrade.
+	// Otherwise, the upgrade will be unsuccessful (see
+	// https://github.com/elastic/elastic-agent/issues/2645).
+
+	// If the installation marker file is present, we're all set.
+	if info.RunningInstalled() {
+		return nil
+	}
```

Comment:
I think adding a comment saying that only installed Elastic Agent's can be self-upgraded so if this returns true then we know that the marker is present as `info.RunningInstalled()` checks for the marker to be present.

Reply:
Done in 4a420fa99.